src/Sfa.Das.Sas.Infrastructure/CourseDirectory/CourseDirectoryProviderMapper.cs

```diff
@@ -43,6 +43,33 @@ public Provider Map(CourseDirectoryProvider input)
             return providerImport;
         }
 
+        public Provider PopulateExistingProviderFromCD(CourseDirectoryProvider input, Provider prov)
+        {
+            var providerLocations = input.Locations.Select(MapToLocationEntity);
+
+            prov.Id = input.Id.ToString();
+
+            prov.Name = !string.IsNullOrEmpty(input.Name) ? input.Name : prov.LegalName;
```

Comment:
You don't need check if the name is empty just if it's on the course directory or not.